src/serprog.c

```diff
@@ -482,14 +481,18 @@ static int serprog_parseextparams(const PROGRAMMER *pgm, const LISTID extparms)
     }
 
     if(str_eq(extended_param, "help")) {
-      msg_error("%s -c %s extended options:\n", progname, pgmid);
-      msg_error("  -x cs=<n> Sets the chip select line to CS<n> for supported programmers\n");
-      msg_error("  -x help   Show this help menu and exit\n");
+      help = true;
```

Comment:
The next line needs to be `rv = LIBAVRDUDE_EXIT;` not `return ...`

I recommend to checking all 22 treated programmers by running
 ```
avrdude -c ... -x invalid
avrdude -c ... -x help
```
to see whether there are surprises.